tests/integration/targets/rds_cluster_restore/tasks/main.yaml

```diff
@@ -47,15 +47,15 @@
       - "'engine' in _result_create_source_db_cluster"
       - _result_create_source_db_cluster.engine == "{{ engine }}"
       - "'engine_mode' in _result_create_source_db_cluster"
-      - _result_create_source_db_cluster.engine_mode == "provisioned"
+      - _result_create_source_db_cluster.engine_mode == "serverless"
```

Comment:
@alinabuzachis this is not related to this PRs changes, however CI integration tests were failing on this
after debugging a bit, found out the the task above this which was creating the cluster had been creating cluster with `engine_mode = serverless` always.

Reply:
Aurora Serverless is a sort of a different AWS service than RDS, but it uses the RDS API.  I'm not totally sure why this has suddenly changed but `aurora` isn't even listed as an engine type anymore in https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBCluster.html

Since `serverless engine mode only applies for Aurora Serverless v1 DB clusters` we probably don't want to switch to this, since there's already a v2 out.

We might want to update the engine type in tests/integration/targets/rds_cluster_restore/defaults/main.yml instead, to use `aurora-mysql` instead of just `aurora`.  That seems like it would keep the current behaviour of deploying a DB with provisioned infrastructure.  I'm not certain for sure though.